avr-specs/sync-from-upstream.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import subprocess
+import json
+import copy
+import os
+
+SPECS = {
+    "atmega32u4": {
+        "cpu": "atmega32u4",
+    },
+    "atmega48p": {
+        "cpu": "atmega48p",
+    },
+    "atmega168": {
+        "cpu": "atmega168",
+    },
+    "atmega328p": {
+        "cpu": "atmega328p",
+    },
+    "atmega1280": {
+        "cpu": "atmega1280",
+    },
+    "atmega2560": {
+        "cpu": "atmega2560",
+    },
+    "attiny85": {
+        "cpu": "attiny85",
+    },
+    "attiny88": {
+        "cpu": "attiny88",
+    },
+}
+
+COMMON = {
+    "no-default-libraries": False,
+}
+
+
+def main():
+    rustc_version = subprocess.run(
+        ['rustc', '--version'],
+        check=True,
+        stdout=subprocess.PIPE,
+    ).stdout.decode()
+
+    if 'nightly' not in rustc_version:
+        raise Exception('You need nightly rustc!')
+
+    upstream_spec_string = subprocess.run(
+        ['rustc', '--print', 'target-spec-json', '-Z',
+            'unstable-options', '--target', 'avr-unknown-gnu-atmega328'],
+        check=True,
+        stdout=subprocess.PIPE,
+    ).stdout
+
+    if upstream_spec_string is not None:
+        upstream_spec = json.loads(upstream_spec_string)
```

Comment:
Starting to look really nice, thanks for working on this!

One more note here:  What if `upstream_spec_string` is `None`?  Then `upstream_spec` won't be defined and code further down will raise exceptions.  Under what conditions would `upstream_spec_string` be `None` anyway?  The only thing I can think of would be a command failure but that's already guarded by `check=True`.  So I'd just remove the if here:

```suggestion
    upstream_spec = json.loads(upstream_spec_string)
```